test/query.test.js

```diff
@@ -2393,6 +2393,79 @@ describe('query', () => {
     expect(results.data).toHaveLength(20)
   })
 
+  test('reverse', async () => {
+    // Array
+    const numArray = [1, 2, 3]
+
+    // Single page
+    const databases = await client.query(query.Paginate(query.Indexes()))
+    const reverseDBs = await client.query(
+      query.Reverse(query.Paginate(query.Indexes()))
+    )
+
+    // Multi Page
+    const anderson = await client.query(
+      query.Create(query.Collection('widgets'), {
+        data: {
+          name: 'anderson paak',
+          age: 34,
+          n: 100,
+        },
+      })
+    )
+
+    const chance = await client.query(
+      query.Create(query.Collection('widgets'), {
+        data: {
+          name: 'chance the rapper',
+          age: 27,
+          n: 100,
+        },
+      })
+    )
+
+    const kendrick = await client.query(
+      query.Create(query.Collection('widgets'), {
+        data: {
+          name: 'kendrick lamar',
+          age: 32,
+          n: 100,
+        },
+      })
+    )
+
+    const multiPage = await client.query(
+      query.Paginate(query.Match(query.Index('widgets_by_n'), 100), {
+        size: 2,
+        after: [anderson.ref],
+      })
+    )
+
+    const reverseMultiPage = await client.query(
+      query.Reverse(
+        query.Paginate(query.Match(query.Index('widgets_by_n'), 100), {
+          size: 2,
+          after: [anderson.ref],
+        })
+      )
+    )
+
+    // Set
+    const documentsSet = await client.query(
+      query.Paginate(query.Documents(query.Collection('widgets')))
+    )
+    const reverseDocumentsSet = await client.query(
+      query.Reverse(
+        query.Paginate(query.Documents(query.Collection('widgets')))
```

Comment:
if you are calling paginate first, then you are reversing a page, not the set

Reply:
@marrony Thanks for the note 🙏 Instead, should I do `Paginate(Reverse(Documents(Collection(...))))`?

Reply:
yes